### PR TITLE
feat(headless): add support for anonymous analytics events 

### PR DESCRIPTION
**Pull request by olamothe**
Following this PR https://github.com/coveo/coveo.analytics.js/pull/291

This means users will now be able to configure their analytics to specify that it should be "anonymous", either on initial engine configuration, or through a dedicated `setAnonymous` action.

https://coveord.atlassian.net/browse/KIT-856

### CODE REVIEW

File: packages/headless/src/app/engine-configuration.ts [hunk 1]
```diff
@@ -86,6 +86,11 @@ export interface AnalyticsConfiguration {
    * See https://github.com/coveo/coveo.analytics.js for more info.
    */
   runtimeEnvironment?: AnalyticsRuntimeEnvironment;
+  /**
+   * Whether the interaction that caused the search interface to log the event was triggered by an anonymous user.
```
Comment (samisayegh) [inline packages/headless/src/app/engine-configuration.ts:90]: Suggestion: Whether analytics events should be logged anonymously.

File: packages/headless/src/features/configuration/configuration-actions-loader.ts [hunk 2]
```diff
@@ -100,5 +111,6 @@ export function loadConfigurationActions(
     setOriginLevel3,
     updateAnalyticsConfiguration,
     updateBasicConfiguration,
+    setAnonymous,
```
Comment (samisayegh) [inline packages/headless/src/features/configuration/configuration-actions-loader.ts:114]: Is a dedicated action needed? Or can we leverage the `updateAnalyticsConfiguration` action?

File: packages/headless/src/features/configuration/configuration-slice.ts [hunk 3]
```diff
@@ -72,4 +77,7 @@ export const configurationReducer = createReducer(
       .addCase(setOriginLevel3, (state, action) => {
         state.analytics.originLevel3 = action.payload.originLevel3;
       })
+      .addCase(setAnonymous, (state, action) => {
+        state.analytics.anonymous = action.payload.anonymous;
+      })
```
Comment (samisayegh) [inline packages/headless/src/features/configuration/configuration-slice.ts:82]: Is there a reason to have two different ways to set this property? If not, I would suggest just using `updateAnalyticsConfiguration`

**Review by samisayegh — APPROVED** [review]

File: packages/headless/src/features/configuration/configuration-slice.ts (hunk 3 above)
Comment (olamothe) [reply packages/headless/src/features/configuration/configuration-slice.ts:82]: That's similar to all the other analytics property in this file, which follows the same pattern.

You are right that it's not absolutely needed, and all other similar analytics action could in theory be removed ( `disableAnalytics`, `enableAnalytics`, `setOriginLevel2`, `setOriginLevel3`, etc). I'm not sure I remember why they were created in the first place.

We can't do this without a breaking change, however.

Which do you prefer ?

1- I don't add a new function for this, going against what we do for other analytics properties
2- I add one, even if it's repetitive.

File: packages/headless/src/features/configuration/configuration-actions-loader.ts (hunk 2 above)
Comment (olamothe) [reply packages/headless/src/features/configuration/configuration-actions-loader.ts:114]: Answered below.

File: packages/headless/src/features/configuration/configuration-slice.ts (hunk 3 above)
Comment (samisayegh) [reply packages/headless/src/features/configuration/configuration-slice.ts:82]: I prefer to have a single approach, and not propagate patterns unless there is a good reason to do so.
We can also deprecate any redundant actions, log warnings when they are dispatched, and eventually remove them as part of v2.

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**